### ISSUE TITLE
Choosing which canvas to use with the `wasm-experimental` feature

### ISSUE THREAD

**Issue by paolobettelini** [nannou-org/nannou#930]
Hello,

is it possible to pass the canvas I want to draw onto as a `JsValue` or get it with its ID?

I tried modifying the code such that it uses an `HtmlCanvasElement` I give it rather than using `window.canvas()`in the `build_async` function, but upon running it in the browser it said `Canvas is not found`.

Thanks a lot

**Comment by merlindru** [comment]
Quick question, how did you get Nannou to add a canvas to the end of a page?

**Comment by paolobettelini** [comment]
Yes, I reckon I used this template [template](https://github.com/tomoyanonymous/nannou-web-template).

**Comment by merlindru** [comment]
@paolobettelini [JulianCataldo/astro-nannou-starter](https://github.com/JulianCataldo/astro-nannou-starter) solves this by embedding the nannou app in an iframe, like this:

```astro
<iframe src="./nannou-embed" />
```

nannou-embed:

```astro
<script>
	await import("../../crates/my-app/pkg/nannou_web_test.js").then(
		async ({ default: nannou }) => {
			await nannou().then((s: any) => {
				s.main_web();
			});
		}
	);
</script>
```

**Comment by paolobettelini** [comment]
Yes, for now I'm also embedding them in an iframe but it isn't really a clean solution and I would like to directly interact with multiple nannou objects on the same page (like: don't render if not visible, send a signal of some kind). Those are features I can''t quite get yet. Thank you that repository I will check it out
